src/common/base-store.ts

```diff
@@ -73,7 +75,9 @@ export class BaseStore<T = any> extends Singleton {
 
   enableSync() {
     this.syncDisposers.push(
-      reaction(() => this.toJSON(), model => this.onModelChange(model)),
+      reaction(() => this.toJSON(), model => this.onModelChange(model), {
+        delay: this.params.syncDelayMs,
```

Comment:
Why do we need a delay now?

Reply:
It was added to avoid possible infinite cycle of saving store.json in some cases when update comes from renderer
(don't remember exact case, usually hard to reproduce)

Reply:
I ran into something like that too. The fix that I used was making sure any updates happen within the store class and are marked as `@action` that way multiple deep updates don't cause an infinite loop. But if you have tried that this might help